### PR TITLE
GG-36511 Аix concurrentModificationException for ScanQuerySystemView

### PR DESCRIPTION
**Pull request by ygerzhedovich**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java [hunk 1]
```diff
@@ -146,7 +148,11 @@ public QueryDataIterator() {
 
                     reqMap = next.getValue();
 
-                    qriesIter = next.getValue().entrySet().iterator();
+                    Set<Map.Entry<Long, GridFutureAdapter<GridCacheQueryManager.QueryResult<K, V>>>> cpOfEntries =
+                        new HashSet<>(reqMap.entrySet());
+
+                    qriesIter = cpOfEntries.iterator();
+
```
[the remark this answers is not in the document]
Comment (AMashenkov) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java:155]: `new HashSet<>(reqMap.entrySet());`
How does this help to beat concurrent modification exception? 
It looks like just reducing a window when CME could happens.

[the remark this answers is not in the document]
Comment (ygerzhedovich) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java [hunk 2]
```diff
@@ -146,7 +148,13 @@ public QueryDataIterator() {
...
                     reqMap = next.getValue();
 
-                    qriesIter = next.getValue().entrySet().iterator();
+                    Set<Map.Entry<Long, GridFutureAdapter<GridCacheQueryManager.QueryResult<K, V>>>> cpOfEntries;
+                    // It should prevent concurrent modification of the map while a snapshot of the current state is in progress
+                    synchronized (reqMap) {
+                        cpOfEntries = new LinkedHashSet(reqMap.entrySet());
+                    }
+
+                    qriesIter = cpOfEntries.iterator();
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java:157]:
```suggestion
                    Map<Long, GridFutureAdapter<GridCacheQueryManager.QueryResult<K, V>>> cpOfEntries;
                    // It should prevent concurrent modification of the map while a snapshot of the current state is in progress
                    synchronized (reqMap) {
                        cpOfEntries = new HashMap(reqMap);
                    }

                    qriesIter = cpOfEntries.entrySet().iterator();
```

Comment (ygerzhedovich) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java:157]: No, we have ordered a Set and should keep ordered one after copy

**Review by zstan — APPROVED** [review]
thanks, looks good

**Pull request merged**